deploy/apex-client/overlays/dev/kustomization.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: apex
+bases:
+  - ../../base
+secretGenerator:
+  - name: apex-client-secret
+    literals:
+      - cert=<APEX_CONTROLLER_CERT>
+      - url=https://apex.local
+      - auth_url=https://auth.apex.local/token
+      - username=kitteh1
+      - password=floofykittens
+commonLabels:
+  app.kubernetes.io/component: apex-client
+  app.kubernetes.io/instance: apex-client
+  app.kubernetes.io/name: apex-client
+patchesJson6902:
+  - target:
+      kind: DaemonSet
+      name: apex
+    patch: |-
+      - op: replace
+        path: /spec/template/spec/hostAliases/0/ip
+        value: <APEX_CONTROLLER_IP>
```

Comment:
> Since we're in the dev overlay here, we know that the controller is running in the same cluster. I'm not sure how to automate setting the right IP in `hostAliases`. I suppose the init container could do it, instead.
> 
Not sure if that is the ideal case, because installing agent on the same cluster doesn't give us much value, because those nodes are already connected through Kubernetes CNI. Even in the dev case, the agent needs to be deployed in microshift or other cluster nodes to create a mesh? I this in for QA it makes sense to setup this, because we are testing everything in the same Kind cluster.
> If we leave it this way, I would rather not provide instructions that tell people to modify a file checked into git with local values. I would rename this to `kustomization.yaml.sample`, add `kustomization.yaml` to the `.gitignore` for this file, and then ask people to make a local copy of the sample.

Yeah, that makes sense. I think we need to do this across the board in apex/observability/pipeline etc.

Reply:
@russellb created sample overlay and also updated existing dev overlay. PTAL